### PR TITLE
Spring boot hw 2

### PR DESCRIPTION
**Pull request by ShemsheyMaxim**

### CODE REVIEW

File: src/main/java/com/springboot/onlinebanking/model/Currency.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import lombok.Data;
+
+@Data
+@Entity
+@Table(name = "cyrrencies")
```
Comment (KseniiaMakarova) [inline src/main/java/com/springboot/onlinebanking/model/Currency.java]:
```suggestion
@Table(name = "currencies")
```

File: src/main/java/com/springboot/onlinebanking/model/Transaction.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.Data;
+
+@Data
+@Entity
+@Table(name = "transactions")
+public class Transaction {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToOne
```
Comment (KseniiaMakarova) [inline src/main/java/com/springboot/onlinebanking/model/Transaction.java]: why ManyToOne?

File: src/main/java/com/springboot/onlinebanking/model/Transaction.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    @JoinColumn(name = "from_account", nullable = false)
+    private Account fromAccount;
+    @OneToOne
+    @JoinColumn(name = "to_account", nullable = false)
```
Comment (KseniiaMakarova) [inline src/main/java/com/springboot/onlinebanking/model/Transaction.java:26]: same

File: src/main/java/com/springboot/onlinebanking/service/AccountService.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.springboot.onlinebanking.model.Account;
+import java.util.List;
+
+public interface AccountService {
+    Account save(Account account);
+
+    List<Account> findAllUserAccountsByUserPhone(String phoneNumber);
+
+    Account findAccountByAccountNumber(String accountNumber);
```
Comment (KseniiaMakarova) [inline src/main/java/com/springboot/onlinebanking/service/AccountService.java]:
```suggestion
    Account getByAccountNumber(String accountNumber);
```

File: src/main/java/com/springboot/onlinebanking/service/TransactionService.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.springboot.onlinebanking.service;
+
+import com.springboot.onlinebanking.model.Account;
+import com.springboot.onlinebanking.model.Transaction;
+import java.util.List;
+
+public interface TransactionService {
+    List<Transaction> getAllByAccount(int page, int size, Account account);
```
Comment (KseniiaMakarova) [inline src/main/java/com/springboot/onlinebanking/service/TransactionService.java:10]: let's also have a `save()` method

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/springboot/onlinebanking/model/Currency.java (hunk 1 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/springboot/onlinebanking/model/Currency.java]: done

File: src/main/java/com/springboot/onlinebanking/model/Transaction.java (hunk 2 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/springboot/onlinebanking/model/Transaction.java]: done

File: src/main/java/com/springboot/onlinebanking/model/Transaction.java (hunk 3 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/springboot/onlinebanking/model/Transaction.java:26]: done

File: src/main/java/com/springboot/onlinebanking/service/AccountService.java (hunk 4 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/springboot/onlinebanking/service/AccountService.java]: done

File: src/main/java/com/springboot/onlinebanking/service/TransactionService.java (hunk 5 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/springboot/onlinebanking/service/TransactionService.java:10]: done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**